prisma/prisma-engines
# feat(driver-adapters): add fork of "cuid" with `wasm32-unknown-unknown` support

Closes https://github.com/prisma/team-orm/issues/350. Closes https://github.com/prisma/team-orm/issues/282.

This PR:
- uses a [cuid fork](https://github.com/prisma/cuid-rust) that is compatible with `wasm32-unknown-unknown` targets (as long the `js` feature of the `getrandom` dependency is active). See [diff](https://github.com/mplanchard/cuid-rust/compare/master...jkomyno:cuid-rust:jkomyno/wasm32?expand=1).
- removes `#[allow_deprecated]` where no longer required by `cuid`'s fork

## How to Test

- update `query-engine/dmmf/Cargo.toml`:
  ```diff
  [package]
  name = "dmmf"
  version = "0.1.0"
  edition = "2021"
  
  [dependencies]
  bigdecimal = "0.3.0"
  psl.workspace = true
  serde.workspace = true
  serde_json.workspace = true
  schema = { path = "../schema" }
  indexmap = { version = "1.7", features = ["serde-1"] }
  - prisma-models = { path = "../prisma-models" }
  + prisma-models = { path = "../prisma-models", features = ["default_generators"] }
  
  [dev-dependencies]
  expect-test = "1.2.2"
  indoc.workspace = true
  pretty_assertions = "1"
  flate2 = "1.0"
  similar = { version = "2.2.1", features=["text", "inline", "bytes"] }
  colored = "2.0.0"
  itertools = "0.10.5"
  ```
- run `cargo build -p prisma-schema-build --target wasm32-unknown-unknown`
- run `echo $?` (`0` is expected)
